### PR TITLE
Check if kubelet is running in the  initial user namespace.

### PR DESCRIPTION
**Pull request by yanghaichao12**
/kind flake

What this PR does / why we need it:
Check if kubelet is running in the  initial  user namespace.

Which issue(s) this PR fixes (optional, in fixes #(, fixes #<issue_number>, ...) format, will close the issue(s) when PR gets merged):
Fixes #

Special notes for your reviewer:

Does this PR introduce a user-facing change?:

NONE

### CODE REVIEW

File: cmd/kubelet/app/server.go [hunk 1]
```diff
@@ -418,8 +418,15 @@ func checkPermissions() error {
 	if uid := os.Getuid(); uid != 0 {
 		return fmt.Errorf("Kubelet needs to run as uid `0`. It is being run as %d", uid)
 	}
-	// TODO: Check if kubelet is running in the `initial` user namespace.
+	// Check if kubelet is running in the `initial` user namespace.
 	// http://man7.org/linux/man-pages/man7/user_namespaces.7.html
+	namespace, _ := os.Readlink("/proc/" + strconv.Itoa(os.Getpid()) + "/ns/user")
```
Comment (dims) [inline cmd/kubelet/app/server.go]: This will break the windows usecase ..

Comment (yanghaichao12) [reply cmd/kubelet/app/server.go]: you‘re right，i’ve add ostype check，thanks！

**Comment by anarchistHH1983** [comment]
retest

**Comment by derekwaynecarr** [comment]
Thank you for the PR and apologies for the delay.  This had fallen off my radar.  Given that containerized kubelet is now deprecated, I think the need to further evolve this check at this time is no longer pressing.  If/when we add back any future support for kubelet in a user namespace, we can evaluate this PR or a follow-on at that time.

**Pull request closed without merging**